tests/test_update_checks.py

```diff
@@ -0,0 +1,57 @@
+from astronomer.airflow.version_check.models import AstronomerVersionCheck, AstronomerAvailableVersion
+from astronomer.airflow.version_check.update_checks import CheckThread, UpdateAvailableBlueprint
+from unittest import mock
+import pytest
+from packaging import version
+
+
+@pytest.mark.parametrize("image_version, lowest_tag", [("2.2.0.post2", "2.2.0"), ("2.2.3.post1", "2.2.3")])
+@mock.patch('astronomer.airflow.version_check.update_checks.get_ac_version')
+def test_update_check_for_image_with_newer_patch(mock_ac_version, image_version, lowest_tag, app, session):
+    from airflow.utils.db import resetdb
+
+    mock_ac_version.return_value = image_version
+    with app.app_context():
+        resetdb()
+        vc = AstronomerVersionCheck(singleton=True)
+        session.add(vc)
+        session.commit()
+
+        thread = CheckThread()
+        thread.ac_version = image_version
+        thread.check_for_update()
+        # Here we check for the highest patch on the selected image
+        # If the image is 2.2.0.post2, then the highest patch may be 2.2.0-5
+        latest_patch = (
+            session.query(AstronomerAvailableVersion)
+            .filter(AstronomerAvailableVersion.version.like(f"{lowest_tag}%"))
+            .order_by(AstronomerAvailableVersion.date_released.desc())
+            .first()
+        )
+        blueprint = UpdateAvailableBlueprint()
+        result = blueprint.available_update()
+        assert result['version'] == latest_patch.version
+
+
+@mock.patch('astronomer.airflow.version_check.update_checks.get_ac_version')
+def test_update_check_for_image_already_on_the_highest_patch(mock_ac_version, app, session):
+    from airflow.utils.db import resetdb
+
+    image_version = "2.0.0.post10"
```

Comment:
Since the support for 2.0.0 is ending this month, I think there won't be any more patch releases on 2.0.0